src/main/py/peer/peer.py

```diff
@@ -178,11 +180,14 @@ async def announce_trade(self, wait_time=timedelta(seconds=10)):
         offer_announcement = OfferAnnouncement(artwork)
         announcement_key = utils.generate_random_sha1_hash()
         self.inventory.add_pending_trade(announcement_key, offer_announcement)
+
         asyncio.get_event_loop().call_later(
-            wait_time,
-            asyncio.create_task,
-            self.handle_announcement_deadline(announcement_key, offer_announcement),
+            wait_time.total_seconds(),
+            lambda: asyncio.create_task(
+                self.handle_announcement_deadline(announcement_key, offer_announcement)
+            ),
         )
+
         try:
```

Comment:
@thompalex the code in `handle_announcement_deadline` starting with line 159 above and this `try` block seems to be the same... seems like the 159 copy shouldn't be there. Is that the case? If not, what am I missing?

Reply:
Capturing the conversation in our meeting.
* The log messages are a bit misleading.
* Announcing the commission again causes Contributors to evaluate the deadline and detect that it has passed.

I encouraged thinking about how we can make this easier to reason about in the code flow, rather than the implicit behavior. Especially given the async approach, having more explicit behavior will make it easier for others... and yourself in a few months :-)